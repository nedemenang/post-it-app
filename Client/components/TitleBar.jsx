import React, { Component } from 'react';
import '../public/style.css';
import { AppBar } from 'material-ui';
import { Link } from 'react-router-dom';
import { signOutUser } from '../actions/AppActions';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';
import IconButton from 'material-ui/IconButton';
import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';

/**
 * @return {void} return void
 */
function signOut() {
  signOutUser();
  localStorage.removeItem('user');
}

const Menu = () => (

  <IconMenu
    iconButtonElement={
      <IconButton><MoreVertIcon /></IconButton>
    }
    targetOrigin={{ horizontal: 'right', vertical: 'top' }}
    anchorOrigin={{ horizontal: 'right', vertical: 'top' }}
  >
  <MenuItem
            primaryText="home"
            containerElement={<Link to="/#" />}
          />
          <MenuItem
            primaryText="Edit Profile"
            containerElement={<Link to="/ProfileEdit" />}
          />
          <MenuItem
            primaryText="Sign Out"
            containerElement={<Link to="/" />}
            onTouchTap={signOut}
          />
  </IconMenu>
);

/**
 * @class TitleBar
 * @extends {Component}
 */
class TitleBar extends Component {
  /**
   *
   * Renders title bar page
   * @returns {JSX} returns title bar page
   * @memberof TitleBar
   */
  render() {
    return (
      <div className="topnav">
          <AppBar title="Post It App" iconElementRight={< Menu />} />
      </div>
    );
  }


}

export default TitleBar;
