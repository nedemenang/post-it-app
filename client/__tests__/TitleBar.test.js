import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TitleBar from '../components/TitleBar.jsx';

describe('the title bar component', () => {
  let mountedComponent;

  const TitleBarComponent = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<TitleBar />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render', () => {
    const component = TitleBarComponent();
    expect(component).toBeDefined();
  });

  it('should match snapshot test', () => {
    const component = TitleBarComponent();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
