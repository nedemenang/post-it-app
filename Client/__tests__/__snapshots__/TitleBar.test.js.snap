// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the title bar component should match snapshot test 1`] = `
<div
  className="topnav"
>
  <AppBar
    iconElementRight={<Menu />}
    showMenuIconButton={true}
    title="Post It App"
    zDepth={1}
  />
</div>
`;
